Code/GraphMol/Subgraphs/Subgraphs.cpp

```diff
@@ -590,14 +597,29 @@ PATH_TYPE findAtomEnvironmentOfRadiusN(const ROMol &mol, unsigned int radius,
     }
     nbrStack = nextLayer;
   }
-  if (i != radius) {
-    // this happens when there are no paths with the requested radius.
-    // return nothing in this case:
+  if (i != radius && enforceSize) {
+    // If there are no paths found with the requested radius, user can choose whether 
+    // or not to return nothing in this case. If enforceSize=true, this is similar to
+    // the previous bahviour (return an empty path/vector). Otherwise, it collect every 
+    // path within the requested radius. This is similar to maxPath(mol, res) <= radius.
+    // The atomMap would have similar behaviour as bondPath.
     res.clear();
     res.resize(0);
+    atomMap.clear();
   }
-
   return res;
 }
 
+PATH_TYPE findAtomEnvironmentOfRadiusN(
+    const ROMol &mol, unsigned int radius, unsigned int rootedAtAtom,
+    bool useHs, bool enforceSize) {
+  // Overiding function for backward compatibility
+  std::unordered_map<unsigned int, unsigned int> atomMap;
+  PATH_TYPE pth = findAtomEnvironmentOfRadiusN(mol, radius, rootedAtAtom, 
+                                               useHs, enforceSize, atomMap);
+  atomMap.clear();
```

Comment:
this call to `clear()` is unnecessary. The local variable will be cleaned up when the function returns.

Reply:
I am just worried about memory leak. Thank you

Reply:
> I am just worried about memory leak. Thank you

It's always good to worry about memory leaks :-)

In this case you don't have to worry because atomMap is a local variable and C++ guarantees you that local variables will be cleaned up when they go out of scope.